tests/testthat/test-scale-discrete.R

```diff
@@ -16,3 +16,19 @@ test_that("discrete ranges also encompas continuous values", {
   expect_equal(x_range(base + geom_point(aes(x1)) + geom_point(aes(x2))), c(0, 4))
 })
 
+test_that("discrete scale shrinks to range when setting limits", {
+  p <- ggplot(subset(diamonds, carat > 1), aes(cut, clarity)) +
+    geom_jitter() + scale_x_discrete(limits=c("Fair","Ideal"))
+
+  expect_equal(layer_scales(p)$x$dimension(c(0, 1)), c(0, 3))
+})
+
+test_that("limits are returned correctly when using continuous input", {
+  xyDF <- data.frame(x = 1:10, y = 1:10)
+  p <- ggplot(data = xyDF, mapping = aes(x = x, y = y)) +
+    geom_line() +
+    geom_point() +
+    scale_x_discrete()
+
+  expect_equal(layer_scales(p)$x$get_limits(), seq_len(10))
```

Comment:
Calculations of breaks are not really affected by this... If you plot a huge continuous range on a discrete scale it will create a lot of breaks...